frontend/src/components/general-comparison-table/general-comparison-table.tsx

```diff
@@ -0,0 +1,128 @@
+import { useLayoutEffect, useState } from 'react';
+
+import { CollapseElement } from '@components/collapse-component/collapse-element/collapse-element';
+import { Spinner } from '@components/common/spinner/spinner';
+import { getElementHeightWithMargins } from '@helpers/utils/get-element-height-with-margins';
+import { useGetComparisonGeneralInfoQuery } from '@store/queries/comparison';
+import { clsx } from 'clsx';
+
+import styles from './styles.module.scss';
+
+const GeneralComparisonTable: React.FC = () => {
+  const { data: generalInfos, isLoading } = useGetComparisonGeneralInfoQuery();
+
+  const options: Set<string> = new Set();
+  generalInfos?.forEach((car) =>
+    Object.keys(car.options).map((optionType: string) =>
+      options.add(optionType),
+    ),
+  );
+
+  const [generalTable, setGeneralTableRef] = useState<HTMLDivElement | null>(
+    null,
+  );
+
+  useLayoutEffect(() => {
+    if (!generalTable) return;
+
+    const optionsValues = generalTable.querySelectorAll(
+      '[data-optionvalue]',
+    ) as NodeListOf<HTMLDivElement>;
+
+    const optionsHighestHeights = new Map();
+
+    [...optionsValues].forEach((value: HTMLDivElement) => {
+      const borderHeight = 1;
+      const height = getElementHeightWithMargins(value) + borderHeight;
+      const title = value.getAttribute('data-optionvalue');
+
+      if (
+        height > optionsHighestHeights.get(title) ||
+        !optionsHighestHeights.has(title)
+      ) {
+        optionsHighestHeights.set(title, height);
+      }
+    });
+
+    [...optionsValues].forEach((value: HTMLDivElement) => {
+      const title = value.getAttribute('data-optionvalue');
+      const tableTitleElement = generalTable.querySelector(
+        '[data-optiontitle=' + title + ']',
+      ) as HTMLDivElement;
+      value.style.height = `${optionsHighestHeights.get(title)}px`;
+      tableTitleElement.style.height = `${optionsHighestHeights.get(title)}px`;
+    });
+  }, [generalTable]);
```

Comment:
Could you please sprinkle this code with some comments? Because it's not that obvious what is happening here :sweat_smile:

Reply:
I added the comments. Hope the code became more readable now. It's functionality to make table cells equal height. 
Any additional simplification advice would be appreciated :)